flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1400,6 +1411,23 @@ public TableResultInternal executeInternal(Operation operation) {
             } catch (Exception e) {
                 throw new TableException("Failed to execute ANALYZE TABLE command", e);
             }
+        } else if (operation instanceof DeleteFromFilterOperation) {
+            if (isStreamingMode) {
+                throw new TableException("DELETE TABLE is not supported for streaming mode now.");
+            }
+            Optional<Long> rows =
```

Comment:
This implement has some limit:
1. Predicate cannot contains subquery, it must be a deterministic expressions.
2. This delete or update runtime execution is at client environment, maybe it's network environment has a littele bit different from taskmanager in cluster.(e.g. taskmanager can connect data source, but client cannot connect it , so execute in here may fail) 

Do you think those limitations are acceptable or not?

Reply:
Thanks for attention.
Yes, it's acceptable.
1: When the predicate contain subquery, the delete can't be pushed,  it can fall back to SupportsRowLevelDelete which support to delete for all cases.
2: Still, only delete can pushed to table source, will we do the deletion in here. If the delete can be pushed down, it's be more like it's to drop some partition/delete files for storage based on files.  I think it's similar to drop partition which is also done in TableEnvironmentImpl.  For other storage like mysql, it expect to delegate the delete stamement to the underlying engine,  I think it's fine that client should have the ability to connect to the underlying engine.
Also, I think it's beneficial that execute it in client direclty which we also reduce the cost of generate/deploy job.

For more detail, you can refer to the [FLIP-282](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=235838061)